lib/ansible/modules/cloud/vmware/vmware_host_tcpip_stacks.py

```diff
@@ -0,0 +1,473 @@
+    - ' - C(vmkernel_gateway) (string): The default gateway address used by this instance.'
+    - ' - C(congestion_algorithm) (string): The TCP congestion control algorithm used by this instance. (default: newreno).'
+    - '   - choices: [newreno, cubic]'
+    - ' - C(num_connections) (int): The maximum number of socket connections that are requested on this instance. (default: 11000).'
+    type: dict
+  vxlan:
+    description:
+    - VXLAN TCP/IP stack configuration.
+    - 'The following parameters can be used:'
+    - ' - C(vmkernel_gateway) (string): The default gateway address used by this instance.'
+    - ' - C(congestion_algorithm) (string): The TCP congestion control algorithm used by this instance. (default: newreno).'
+    - '   - choices: [newreno, cubic]'
+    - ' - C(num_connections) (int): The maximum number of socket connections that are requested on this instance. (default: 11000).'
+    type: dict
+  esxi_hostname:
+    description:
+    - Name of the host system to work with.
+    - This parameter is required if C(cluster_name) is not specified.
+    type: str
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This parameter is required if C(esxi_hostname) is not specified.
+    type: str
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Configure vMotion TCP/IP stack gateway
+  vmware_host_tcpip_stacks:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    vmotion:
+      vmkernel_gateway: 192.168.2.1
+  delegate_to: localhost
+
+- name: Configure vMotion TCP/IP stack
+  vmware_host_tcpip_stacks:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    vmotion:
+      vmkernel_gateway: 192.168.2.1
+      congestion_algorithm: newreno
+      num_connections: 11000
+  delegate_to: localhost
+
+- name: Configure all TCP/IP stacks
+  vmware_host_tcpip_stacks:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    default:
+      vmkernel_gateway: 192.168.1.1
+      congestion_algorithm: newreno
+      num_connections: 11000
+    provisioning:
+      congestion_algorithm: newreno
+      num_connections: 11000
+    vmotion:
+      vmkernel_gateway: 192.168.2.1
+      congestion_algorithm: newreno
+      num_connections: 11000
+    vxlan:
+      congestion_algorithm: newreno
+      num_connections: 11000
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+result:
+  description: metadata about host TCP/IP stack configuration
+  returned: always
+  type: dict
+  sample: {
+    "esx01.example.local": {
+      "changed": false,
+      "default_congestion_algorithm": "newreno",
```

Comment:
Curious that the inputs are dictionaries, but the outputs are not.

Reply:
https://github.com/ansible/ansible/pull/49087/files#diff-52826cd185077e0bbf48084382157ba2R44

Reply:
So on the input side, you have default: { ... }, vmotion: { ... }, ... but on the output side you have a flat space with "default_*" and "vmotion_*" names.  Shouldn't the output have the same nesting as the input?

Reply:
I agree with @jeking3 here, it makes more sense to have the return be: 
```json
"hostname.example.local": {
    "changed": false,
    "default": {
        "congestion_algorithm": "newreno",
        "num_connections": 11000,
        "vmkernel_gateway": "192.168.1.1"
    },
    "provisioning": {
        "congestion_algorithm": "newreno",
        "num_connections": 11000,
        "vmkernel_gateway": null
    },
    "vmotion": {
        "congestion_algorithm": "newreno",
        "num_connections": 11000,
        "vmkernel_gateway": null
    },
    "vxlan": {
        "congestion_algorithm": "newreno",
        "num_connections": 11000,
        "vmkernel_gateway": null
    }
}
```
(btw, your output example doesnt have vMotion network info

Reply:
@Akasurde, as the main maintainer, what's your view on this? So we can close the discussion, i think for me this is the only thing i'd really feel stronger about, the refactoring or early failure is less important to me and are nice to have's.

Reply:
What you just said is the same as saying it is okay to add technical debt to the project.  There's no excuse for repeating those large code blocks.  It impacts future sustainability.